src/battle-dex.ts

```diff
@@ -601,45 +613,43 @@ const Dex = new class implements ModdedDex {
 
 		if (animationData[facing + 'f'] && options.gender === 'F') facing += 'f';
 		let allowAnim = !Dex.prefs('noanim') && !Dex.prefs('nogif');
-		if (allowAnim && genNum >= 6) spriteData.pixelated = false;
-		if (allowAnim && animationData[facing] && genNum >= 5) {
+		if (allowAnim && spriteData.gen >= 6) spriteData.pixelated = false;
+		if (allowAnim && animationData[facing] && spriteData.gen >= 5) {
 			if (facing.slice(-1) === 'f') name += '-f';
-			dir = gen + 'ani' + dir;
+			dir = baseDir + 'ani' + dir;
 
 			spriteData.w = animationData[facing].w;
 			spriteData.h = animationData[facing].h;
 			spriteData.url += dir + '/' + name + '.gif';
 		} else {
 			// There is no entry or enough data in pokedex-mini.js
 			// Handle these in case-by-case basis; either using BW sprites or matching the played gen.
-			if (gen === '') gen = 'gen5';
-			dir = gen + dir;
+			dir = (baseDir || 'gen5') + dir;
 
 			// Gender differences don't exist prior to Gen 4,
 			// so there are no sprites for it
-			if (genNum >= 4 && miscData['frontf'] && options.gender === 'F') {
+			if (spriteData.gen >= 4 && miscData['frontf'] && options.gender === 'F') {
 				name += '-f';
 			}
 
 			spriteData.url += dir + '/' + name + '.png';
 		}
 
 		if (!options.noScale) {
-			if (fieldGenNum > 5) {
+			if (graphicsGen > 4) { // TODO: should this be spriteData.gen?
```

Comment:
Is this a graphics level thing or a sprite level thing? I *think* its fine as is, but I just want to confirm.

Reply:
I'd probably make this a graphicsGen thing.